### ISSUE TITLE
Bug in Random for ranges?

### ISSUE THREAD

**Issue by Stefan-Kohl** [gap-system/gap#772]
I observe the following (GAP 4.8.3, 64bit, Ubuntu):

 ```
    gap> Random([1..2^32-1]);
    3151410335
    gap> Random([1..2^32-1]);
    194975213
    gap> Random([1..2^32-1]);
    2707735144
    gap> Random([1..2^32-1]);
    2345474764
    gap> Random([1..2^32-1]);
    2165862404
    gap> Random([1..2^32]);  
    1
    gap> Random([1..2^32]);
    2
    gap> Random([1..2^32]);
    2
    gap> Random([1..2^32]);
    2
    gap> Random([1..2^32]);
    1

```

**Comment by ChrisJefferson** [comment]
Code in FuncRandomListMT was never updated for 64-bit OSes.

I don't really understand why this method exists, why we don't just do this at the GAP level...

**Comment by frankl-audio** [comment]
I vaguely remember that there were some discussions about the code related to Random. This never lead to some decisions and some code was left unfinished. 

Two more problems in addition to the one reported above:
`RandomList` does not use the Mersenne twister as stated in the documentation.
Instead it is identical to the old `RANDOM_LIST`, compiled code from lib/random.g.

Actually, the code for `RANDOM_LIST` also assumes that lists have less than 2^28 entries,
although it is less apparent:
```
 gap> l := [1..2^59];;
gap> RandomList(l);
551029243773976577
gap> RandomList(l);
437149691306573825
gap> RandomList(l);
358415813272666113
gap> RandomList(l) mod 2^28;
1
gap> RandomList(l) mod 2^28;
1
gap> RandomList(l) mod 2^28;
1
```
See pull request "Several fixes for Random on long lists in 64-bit GAP"  (#780) for a proposed fix of all these problems.

**Comment by alex-konovalov** [comment]
Fixed in #781 and cherry-picked onto stable-4.8 branch. The fix will appear in GAP 4.8.4.
